pymc3/plots/traceplot.py

```diff
@@ -106,3 +106,137 @@ def traceplot(trace, varnames=None, transform=identity_transform, figsize=None,
+                   grid=False, alpha=0.35, priors=None, prior_alpha=1,
+                   prior_style='--', ax=None):
+    """Plot samples histograms and values.
+
+    Parameters
+    ----------
+
+    iter_sample : trace generator
+    skip_first : int
+        Number of first samples not shown in plots (burn-in). This affects
+        frequency and stream plots.
+    refresh_every : int
+        Period of plot updates (in sample number)
+    roll_over : int
+        Width of the sliding window for the sample stream plots: last roll_over
+        samples are shown (no effect on frequency plots).
+    varnames : list of variable names
+        Variables to be plotted, if None all variable are plotted
+    transform : callable
+        Function to transform data (defaults to identity)
+    figsize : figure size tuple
+        If None, size is (12, num of variables * 2) inch
+    lines : dict
+        Dictionary of variable name / value  to be overplotted as vertical
+        lines to the posteriors and horizontal lines on sample values
+        e.g. mean of posteriors, true values of a simulation.
+        If an array of values, line colors are matched to posterior colors.
+        Otherwise, a default red line
+    combined : bool
+        Flag for combining multiple chains into a single chain. If False
+        (default), chains will be plotted separately.
+    plot_transformed : bool
+        Flag for plotting automatically transformed variables in addition to
+        original variables (defaults to False).
+    grid : bool
+        Flag for adding gridlines to histogram. Defaults to True.
+    alpha : float
+        Alpha value for plot line. Defaults to 0.35.
+    priors : iterable of PyMC distributions
+        PyMC prior distribution(s) to be plotted alongside posterior. Defaults
+        to None (no prior plots).
+    prior_alpha : float
+        Alpha value for prior plot. Defaults to 1.
+    prior_style : str
+        Line style for prior plot. Defaults to '--' (dashed line).
+    ax : axes
+        Matplotlib axes. Accepts an array of axes, e.g.:
+
+        >>> fig, axs = plt.subplots(3, 2) # 3 RVs
+        >>> pymc3.traceplot(trace, ax=axs)
+
+        Creates own axes by default.
+
+    Returns
+    -------
+
+    MultiTrace object with access to sampling values
+
+    """
+    for it, trace in enumerate(iter_sample):
+        if it > skip_first:
+            if it == skip_first + 1:
+                ax = traceplot(trace[skip_first:], varnames, transform, figsize,
+                               lines, combined, plot_transformed, grid,
+                               alpha, priors, prior_alpha, prior_style,
+                               ax)
+                if varnames is None:
+                    varnames = get_default_varnames(trace, plot_transformed)
+                x0 = skip_first
+            elif (it - skip_first) % refresh_every == 0:
+                for i, v in enumerate(varnames):
```

Comment:
I haven't looked closely, but do you think it is possible for this code to be factored out and shared with the base `traceplot` code?